compiler/rustc_ast/src/token.rs

```diff
@@ -299,6 +299,10 @@ pub enum TokenKind {
     /// It's recommended to use `Token::(ident,uninterpolate,uninterpolated_span)` to
     /// treat regular and interpolated identifiers in the same way.
     Ident(Symbol, /* is_raw */ bool),
+
+    /// A `k#ident` keyword
+    Keyword(Symbol),
```

Comment:
Actual keywords are Ident, so it seems confusing for a "forced keyword" to be special here.